apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -140,66 +137,50 @@ export const updateAiCustomization = createAsyncThunk(
   }
 );
 
-// This thunk's callback function submits a user chat message to the chat completion endpoint,
-// waits for a chat completion response, and updates the user message state.
-export const submitChatMessage = createAsyncThunk(
-  'aichat/submitChatMessage',
-  async (message: string, thunkAPI) => {
+// This thunk's callback function submits a user's chat content and AI customizations to
+// the chat completion endpoint, then waits for a chat completion response, and updates
+// the user messages.
+export const submitChatContents = createAsyncThunk(
+  'aichat/submitChatContents',
+  async (chatContext: ChatContext, thunkAPI) => {
     const state = thunkAPI.getState() as {lab: LabState; aichat: AichatState};
-    const systemPrompt = (state.lab.levelProperties as AichatLevelProperties)
-      ?.systemPrompt;
-    // TODO: move a check for undefined systemPrompt to AIchatView and throw an error dialog
-    // there if systemPrompt is undefined.
-    if (systemPrompt === undefined) {
-      throw new Error('systemPrompt is undefined');
-    }
+    const aiCustomizations = state.aichat.savedAiCustomizations;
     const storedMessages = state.aichat.chatMessages;
+    const newMessageText = chatContext['userMessage'];
```

Comment:
Nit: is there a reason this variable is here? Just access directly as `chatMessageText: chatContext.userMessage`?